consul-lambda-registrator/extension/extension.go

```diff
@@ -0,0 +1,312 @@
+package extension
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"net"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/hashicorp/go-hclog"
+
+	"github.com/hashicorp/terraform-aws-consul-lambda-registrator/consul-lambda-registrator/proxy"
+	"github.com/hashicorp/terraform-aws-consul-lambda-registrator/consul-lambda-registrator/structs"
+	"github.com/hashicorp/terraform-aws-consul-lambda-registrator/consul-lambda-registrator/trace"
+)
+
+type Config struct {
+	MeshGatewayURI      string        `envconfig:"MESH_GATEWAY_URI" required:"true"`
+	ExtensionDataPrefix string        `envconfig:"EXTENSION_DATA_PREFIX" required:"true"`
+	Datacenter          string        `envconfig:"DATACENTER"`
+	ServiceName         string        `envconfig:"SERVICE_NAME"`
+	ServiceNamespace    string        `envconfig:"SERVICE_NAMESPACE"`
+	ServicePartition    string        `envconfig:"SERVICE_PARTITION"`
+	ServiceUpstreams    []string      `envconfig:"SERVICE_UPSTREAMS"`
+	RefreshFrequency    time.Duration `envconfig:"REFRESH_FREQUENCY" default:"5m"`
```

Comment:
In the RFC, these are prefixed with `CONSUL_`. Is there a reason for removing the prefix? It feels like this is helpful for identifying what the environment variables are used for and ensuring they are unlikely to squash other ones.

Reply:
The `CONSUL` prefix actually gets applied globally here: https://github.com/hashicorp/terraform-aws-consul-lambda/blob/cthain/consul-lambda-extension/consul-lambda-registrator/consul-lambda-extension/main.go#L89

I've made this explicit on another branch to get rid of the magic: https://github.com/hashicorp/terraform-aws-consul-lambda/blob/cthain/lambda-example/consul-lambda-registrator/extension/extension.go#L22-L29

Reply:
I've pulled my change into this branch. The env var names are now all prefixed with `CONSUL_` for claritiy and consistency with the RFC.